### ISSUE TITLE
Board reboots after connecting two spi lines.

### ISSUE THREAD

**Issue by GiovanniRaseraF** [dedalqq/esp32-mcp2515#7]
On the circuit I made, the mcp and gps module communicate over the same spi line, but the cs pins are separate. After pulling data over MCP, when I do digitalWrite("mcp' cs pin", HIGH), it goes into SPI_FLASH_BOOT mode and boots continuously.

At the same time, when I run the code without making any connections, the DATA part is empty and the ID part is written. Is it normal? I'll be glad, if you help me

_Originally posted by @MehmetAliAksoy1 in https://github.com/dedalqq/esp32-mcp2515/issues/4#issuecomment-1454732263_

**Comment by GiovanniRaseraF** [comment]
hi  @MehmetAliAksoy1 please post the code you think is important here, so we can try to help you :)

**Comment by MehmetAliAksoy1** [comment]
<img width="445" alt="Ekran Resmi 2023-03-05 01 11 01" src="https://user-images.githubusercontent.com/90469212/222930857-a661c70b-8f82-4044-a2fa-b55edd6dc745.png">

i only added two lines to your example code. The reason I added was to make mcp's cs pin HIGH and cut off communication, gps' cs pin to LOW and open it to communication. When i add this like that, on serial monitor SPI_FLASH_BOOT_MODE writing repeatedly

**Comment by MehmetAliAksoy1** [comment]
Actually i want to select cs pin and take data from gps and spi

**Comment by MehmetAliAksoy1** [comment]
how can i add gps's spi pin to same spi channel? please help
@GiovanniRaseraF

**Comment by GiovanniRaseraF** [comment]
Check some things: 
- you are setting up MISO MOSI CS CLK correctly, this means you are using the write pins. for example for the esp32 800L I'm using this pins: 
```cpp
    #define MISO  19
    #define MOSI  23
    #define SCK   18
    #define SS    5
```
- check the PINOUT of your board if you don't know the pins to chose,
- check if you are compiling for the write target, not all esp32 versions are the same,
- if you have I2C communication on the gps try that to see if it works independently.

**Comment by MehmetAliAksoy1** [comment]
Pins are correct, but i want to select CS pin when i use

**Comment by GiovanniRaseraF** [comment]
Okay, please try all the things and report your findings.
And please provide the name of the gps that you are using.

**Comment by MehmetAliAksoy1** [comment]
#include <SPI.h>

#define GPS_CS 26

void spi_read(char* data) {
  digitalWrite(GPS_CS, LOW);
  delay(10);
  for (int i = 0; i < 100; i++) {
    data[i] = SPI.transfer(0x00);
  }
  digitalWrite(GPS_CS, HIGH);
}

void setup() {
  Serial.begin(9600);
  SPI.begin(14, 12, 13, 26); //sck, miso, mosi, ss
  pinMode(GPS_CS, OUTPUT);
  digitalWrite(GPS_CS, HIGH);
}

void loop() {
  char gps_data[100];
  spi_read(gps_data);
  Serial.println(gps_data);
  delay(1000);
}

This gps code working and print datas. Gps module is Neo-m8. 


When I combine this code with the CAN code, I cannot receive data from GPS. Even when I turn the MCP's CS pin high, data starts coming directly from the can.

**Comment by MehmetAliAksoy1** [comment]
When i call SPI.begin() after The canbus integration at Setup, it is working, why? @GiovanniRaseraF

**Comment by GiovanniRaseraF** [comment]
I think the problem is that you are using SPI in the wrong way, i cannot test it so i cannot be sure.
And i think that Neo-m8 uses I2C at 9600 baud.

The best thing i can tell you is to:
- Create a sketch with only canbus working
- Create a sketch with GPS Neo-m8 using I2C communication ( for reference use this tutorial [GPS tutorial](https://quadmeup.com/how-to-connect-gps-to-esp32/))
- Create a third sketch with the combined code

And at the end post the code in you git and link it here or post it here with a markdown tag; it will be easier to read

**Comment by MehmetAliAksoy1** [comment]
No, we made connection with both of them on HSPI. But gps's select pin 26. I can not use i2c

**Comment by MehmetAliAksoy1** [comment]
And also your code does not work for me now

**Comment by MehmetAliAksoy1** [comment]
I solved the problem but i did not any connection to mcp2515 from read the data. But on the serial monitor: ID: 447 DATA: ID: 42C DATA: ID: 50A DATA: ID: 12E DATA: ID: 341. What is this data came from? How is it possible? @GiovanniRaseraF

**Comment by MehmetAliAksoy1** [comment]
There is no change when SS pin select another free pin.

**Comment by Cyberko102** [comment]
> Я решил проблему, но у меня не было никакого подключения к mcp2515 при чтении данных. А вот на последовательном мониторе: ID: 447 DATA: ID: 42C DATA: ID: 50A DATA: ID: 12E DATA: ID: 341. Откуда взялись эти данные? Как это возможно?@GiovanniRaseraF

Could you show me your code for can bus?

**Comment by Cyberko102** [comment]
> I solved the problem but i did not any connection to mcp2515 from read the data. But on the serial monitor: ID: 447 DATA: ID: 42C DATA: ID: 50A DATA: ID: 12E DATA: ID: 341. What is this data came from? How is it possible? @GiovanniRaseraF

Could you show me your code for can bus?
